scripts/get_auction_elements.ts

```diff
@@ -0,0 +1,144 @@
+import BN from "bn.js";
+import { getOrdersPaginated } from "../src/onchain_reading";
+import type { Order } from "../src/encoding";
+
+const BatchExchange = artifacts.require("BatchExchange");
+const argv = require("yargs")
+  .option("expired", {
+    type: "boolean",
+    describe: "Also show expired orders",
+  })
+  .option("covered", {
+    type: "boolean",
+    describe: "Only show orders that have some balance in the sell token",
+  })
+  .option("tokens", {
+    describe: "Filter only orders between the given tokens",
+  })
+  .option("pageSize", {
+    default: 100,
+    describe: "The page size for the function getOrdersPaginated",
+  })
+  .version(false).argv;
+
+const COLORS = {
+  NONE: "\x1b[0m",
+  RED: "\x1b[31m",
+  YELLOW: "\x1b[33m",
+};
+
+const formatAmount = function (amount: BN) {
+  const string = amount.toString();
+  if (string.length > 4) {
+    return `${string.substring(0, 2)} * 10^${string.length - 2}`;
+  } else {
+    return string;
+  }
+};
+
+const colorForValidFrom = function (order: Order<BN>, currentBatchId: number) {
+  let color = COLORS.NONE;
+  if (order.validFrom >= currentBatchId) {
+    color = COLORS.RED;
+    if (order.validFrom - 5 <= currentBatchId) {
+      color = COLORS.YELLOW;
+    }
+  }
+  return color;
+};
+
+const colorForValidUntil = function (order: Order<BN>, currentBatchId: number) {
+  let color = COLORS.NONE;
+  if (order.validUntil - 5 <= currentBatchId) {
+    color = COLORS.YELLOW;
+    if (order.validUntil <= currentBatchId) {
+      color = COLORS.RED;
+    }
+  }
+  return color;
+};
+
+const colorForRemainingAmount = function (order: Order<BN>) {
+  if (
+    order.priceDenominator.gt(new BN(0)) &&
+    order.remainingAmount
+      .mul(new BN(100))
+      .div(order.priceDenominator)
+      .lt(new BN(1))
+  ) {
+    return COLORS.YELLOW;
+  } else {
+    return COLORS.NONE;
+  }
+};
+
+const printOrder = function (order: Order<BN>, currentBatchId: number) {
+  /* eslint-disable no-console */
+  console.log("{");
```

Comment:
should this take a logger now that we have it?

Reply:
It doesn’t look good with all of the time stamps and info. Just seems unnecessarily wide with the logger.  Also can’t be copied easily... not that we would ever use it.